experimental/src/Config/ConfigInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Experimental\Config;
+
+interface ConfigInterface
+{
+    public function __construct(array $userConfig, array $environmentConfig);
```

Comment:
What's the point of having an interface with a constructor (and only a constructor)?